ui/components/app/snaps/snap-ui-radio-group/snap-ui-radio-group.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import { useSnapInterfaceContext } from '../../../../contexts/snaps';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import {
+  Box,
+  HelpText,
+  HelpTextSeverity,
+  Label,
+  Text,
+} from '../../../component-library';
+
+export type SnapUIRadioProps = {
+  name: string;
+  label?: string;
+  error?: string;
+  options: { name: string; value: string }[];
+  form?: string;
+};
+
+export const SnapUIRadioGroup: FunctionComponent<SnapUIRadioProps> = ({
+  name,
+  label,
+  error,
+  form,
+  ...props
+}) => {
+  const { handleInputChange, getValue } = useSnapInterfaceContext();
+
+  const initialValue = getValue<string>(name, form);
+
+  const [value, setValue] = useState(initialValue ?? '');
+
+  type RadioOptions = [{ value: string; name: string }];
```

Comment:
This is a tuple of only one element, this doesn't look necessary.

I suggest moving RadioOptions outside the function and using it in props.

```typescript
export type SnapUIRadioOption = {value: string, name: string};

export type SnapUIRadioProps = {
  options: SnapUIRadioOption[]
}

export const SnapUIRadioGroup: FunctionComponent<SnapUIRadioProps> = ({
  name,
  label,
  error,
  form,
  ...props
}) => {
  const displayRadioOptions = (options: SnapUIRadioOption[]) => {
    options.map((options) => {}); // this should work automatically
    // but if type is necessary in lambda
    options.map((options: SnapUIRadioOption) => {});
  }
}
```

Reply:
Refactored.